core-web/apps/dotcms-ui/src/app/portlets/shared/dot-content-types-edit/components/dot-binary-settings/dot-binary-settings.component.ts

```diff
@@ -0,0 +1,200 @@
+import { InputSwitchModule } from 'primeng/inputswitch';
+import { InputTextModule } from 'primeng/inputtext';
+
+import { catchError, tap, take } from 'rxjs/operators';
+
+import { DotDialogActions } from '@components/dot-dialog/dot-dialog.component';
+import { DotHttpErrorManagerService, DotMessageService } from '@dotcms/data-access';
+import { DotCMSContentTypeField, DotFieldVariable } from '@dotcms/dotcms-models';
+import { DotMessagePipe } from '@dotcms/ui';
+
+import { DotFieldVariablesService } from '../fields/dot-content-type-fields-variables/services/dot-field-variables.service';
+
+@Component({
+    selector: 'dot-binary-settings',
+    standalone: true,
+    imports: [
+        NgFor,
+        FormsModule,
+        ReactiveFormsModule,
+        InputTextModule,
+        InputSwitchModule,
+        DividerModule,
+        DotMessagePipe
+    ],
+    templateUrl: './dot-binary-settings.component.html',
+    styleUrl: './dot-binary-settings.component.scss',
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DotBinarySettingsComponent implements OnInit, OnChanges {
+    @Input() field: DotCMSContentTypeField;
+    @Input() isVisible: boolean = false;
+
+    @Output() changeControls = new EventEmitter<DotDialogActions>();
+    @Output() valid = new EventEmitter<boolean>();
+    @Output() save = new EventEmitter<DotFieldVariable[]>();
+
+    form: FormGroup;
+
+    private fb: FormBuilder = inject(FormBuilder);
+    private fieldVariablesService = inject(DotFieldVariablesService);
+    private dotMessageService = inject(DotMessageService);
+    private dotHttpErrorManagerService = inject(DotHttpErrorManagerService);
+
+    settingsMap = {
+        accept: {
+            key: 'accept',
+            variable: null
+        },
+        systemOptions: {
+            key: 'systemOptions',
+            variable: null
+        }
+    };
+
+    get settings() {
+        return Object.values(this.settingsMap);
+    }
+
+    protected readonly systemOptions = [
+        {
+            key: 'allowURLImport',
+            message: 'binary-field.settings.system.options.allow.url.import'
+        },
+        {
+            key: 'allowCodeWrite',
+            message: 'binary-field.settings.system.options.allow.code.write'
+        },
+        {
+            key: 'allowFileNameEdit',
+            message: 'binary-field.settings.system.options.allow.file.name.edit'
+        }
+    ];
+
+    ngOnChanges(changes: SimpleChanges) {
+        const { isVisible } = changes;
+        if (isVisible?.currentValue) {
+            this.changeControls.emit(this.dialogActions());
+        }
+    }
+
+    ngOnInit(): void {
+        this.form = this.fb.group({
+            accept: '',
+            // This is an object called systemOptions, this is going to the backend as a field variable
+            allowURLImport: false,
+            allowCodeWrite: false,
+            allowFileNameEdit: false
+        });
```

Comment:
We can use this approach so the `SystemOptions` property has its own `FormGroup`

```ts
const form = new FormGroup({
  systemOptions: new FormGroup({
    allowURLImport: new FormControl(''),
    allowFileNameEdit: new FormControl(''),
    allowCodeWrite: new FormControl(''),
  }),
  acceptType: new FormControl('')
});
```